src/sections/dataset/useCheckPublishCompleted.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import { getDatasetLocks } from '../../dataset/domain/useCases/getDatasetLocks' // Adjust the import path as necessary
+import { Dataset } from '../../dataset/domain/models/Dataset'
+import { DatasetRepository } from '../../dataset/domain/repositories/DatasetRepository'
+
+const useCheckPublishCompleted = (
+  publishInProgress: boolean | undefined,
+  dataset: Dataset | undefined,
+  datasetRepository: DatasetRepository
+): boolean => {
+  const [publishCompleted, setPublishCompleted] = useState(false)
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout | null = null
+
+    if (publishInProgress && dataset) {
+      const gotoReleasedPageAfterPublish = async () => {
```

Comment:
I think the name `gotoReleasedPageAfterPublish` is outdated for what the function does.